database/upgrades/16-refactor-postgres.sql

```diff
@@ -47,14 +50,27 @@ INSERT INTO lost_portals SELECT mxid, chat_id, receiver FROM portal WHERE mxid<>
 -- Make mxid column unique (requires using nulls for missing values)
 UPDATE portal SET mxid=NULL WHERE mxid='';
 ALTER TABLE portal ADD CONSTRAINT portal_mxid_unique UNIQUE(mxid);
+-- Drop foreign keys that make CASCADEs prohibitively slow.
+ALTER TABLE message DROP CONSTRAINT message_portal_fkey;
+ALTER TABLE reaction DROP CONSTRAINT reaction_message_fkey;
```

Comment:
It could probably just skip re-adding the constraints 25 lines above rather than delete them here